backend/lambdas/tasks/generate_queries.py

```diff
@@ -40,20 +50,92 @@
 
 @with_logging
 def handler(event, context):
-    deletion_items = get_deletion_queue(event["ExecutionName"])
-    for data_mapper in get_data_mappers():
+    job_id = event["ExecutionName"]
+    deletion_items = get_deletion_queue()
+    manifests_partitions = []
+    data_mappers = get_data_mappers()
+    total_queries = 0
+    for data_mapper in data_mappers:
         query_executor = data_mapper["QueryExecutor"]
         if query_executor == "athena":
-            queries = generate_athena_queries(data_mapper, deletion_items)
+            queries = generate_athena_queries(data_mapper, deletion_items, job_id)
+            if len(queries) > 0:
+                manifests_partitions.append([job_id, data_mapper["DataMapperId"]])
         else:
             raise NotImplementedError(
                 "Unsupported data mapper query executor: '{}'".format(query_executor)
             )
 
         batch_sqs_msgs(queue, queries)
+        total_queries += len(queries)
+    write_partitions(manifests_partitions)
+    return {
+        "GeneratedQueries": total_queries,
+        "DeletionQueueSize": len(deletion_items),
+        "Manifests": list(
```

Comment:
nit: list comprehension probably makes more sense here since you're immediately calling list with the iterator from map